### PR TITLE
Minor implants refactor

### LINKED ISSUE #24375: Uplink implant
Problem Description: If you get an uplink implant from syndicate bundle with implants, it comes empty.

What did you expect to happen: Expected it to have TC.

What happened instead: I think the uplink implant was changed, and now it comes empty and you need to load it with spare TC. But if you get an implant bundle from syndicate bundle, you can't buy any TC to it, because syndicate bundle costs all TC.

Why is this bad/What are the consequences: Useless uplink in the bundle, it should contain TC or bundle contents should be revised.

Steps to reproduce the problem: Become traitor, pray to RNG that you'll get an implant bundle from syndicate bundle, implant yourself with uplink implant, sigh.

Server: Bagil

Revision: Server revision compiled on: 2017-02-23

### PR DESCRIPTION
**Pull request by coiax**
Fixes #24375.

Implant cases and implanters now have an `imp_type` var for making the
implant in New(), rather than individually creating the implant in
New(), which would have worked fine except when you try to create
subclasses of implanters.

Gang implants are special, and thus untouched.

### CODE REVIEW

File: code/game/objects/items/weapons/implants/implantcase.dm [hunk 1]
```diff
@@ -59,30 +60,22 @@
 
 /obj/item/weapon/implantcase/New()
 	..()
+	if(imp_type)
+		imp = new imp_type(src)
```
Comment (Cyberboss) [inline code/game/objects/items/weapons/implants/implantcase.dm]: `Initialize`

File: code/game/objects/items/weapons/implants/implanter.dm [hunk 2]
```diff
@@ -57,22 +58,14 @@
 
 /obj/item/weapon/implanter/New()
 	..()
+	if(imp_type)
+		imp = new imp_type(src)
```
Comment (Cyberboss) [inline code/game/objects/items/weapons/implants/implanter.dm]: Again

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/game/objects/items/weapons/implants/implanter.dm (hunk 2 above)
Comment (coiax) [inline code/game/objects/items/weapons/implants/implanter.dm]: explain to me the difference please? <3

Comment (Cyberboss) [inline code/game/objects/items/weapons/implants/implanter.dm]: https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#prefer-initialize-over-new-for-atoms

File: code/game/objects/items/weapons/implants/implantcase.dm (hunk 1 above)
Comment (optimumtact) [inline code/game/objects/items/weapons/implants/implantcase.dm]: ^

File: code/game/objects/items/weapons/implants/implanter.dm (hunk 2 above)
Comment (optimumtact) [inline code/game/objects/items/weapons/implants/implanter.dm]: ^

**Comment by coiax** [comment]
@optimumtact done.

**Pull request merged**